paddle/fluid/operators/tensorrt_engine_op.cc

```diff
@@ -55,13 +55,14 @@ nvinfer1::Dims Vec2TRT_Dims(const std::vector<int64_t> &shape) {
                     "TensorRT' tensor input requires at least 2 dimensions");
   PADDLE_ENFORCE_LE(shape.size(), 4UL,
                     "TensorRT' tensor input requires at most 4 dimensions");
+
   switch (shape.size()) {
     case 2:
-      return nvinfer1::Dims2(shape[0], shape[1]);
+      return nvinfer1::Dims2(1, shape[1]);
```

Comment:
why 1 here, anything wrong if `batch_size` larger than 1?

Reply:
ITensor of trt is a structure which records the size of the layer input, assume equal to `single_size`.  The real data is stored in `buf`, and the `batch size` is global setted. Each time forward, the trt will take `single_size * batch` size data from buf.